### ISSUE TITLE
Administration LogFile page crash on OpenCms9.5.0 +

### ISSUE THREAD

**Issue by SandrineProusteau** [alkacon/opencms-core#340]
On a new installed OpenCms 9.5.0 (get it on OpenCms9.5.2 too), when I go to the Admin LogFile page, a have this error : 

```
java.lang.NullPointerException
                at java.io.File.<init>(Unknown Source)
                at org.opencms.util.CmsRfsFileViewer.clone(CmsRfsFileViewer.java:157)
                at org.opencms.workplace.tools.workplace.rfsfile.A_CmsRfsFileWidgetDialog.initLogfileViewBean(A_CmsRfsFileWidgetDialog.java:133)
                at org.opencms.workplace.tools.workplace.rfsfile.A_CmsRfsFileWidgetDialog.defineWidgets(A_CmsRfsFileWidgetDialog.java:112)
                at org.opencms.workplace.tools.workplace.rfsfile.CmsRfsFileViewDialog.defineWidgets(CmsRfsFileViewDialog.java:175)
                at org.opencms.workplace.CmsWidgetDialog.initWorkplaceRequestValues(CmsWidgetDialog.java:1514)
                at org.opencms.workplace.CmsWorkplace.initWorkplaceMembers(CmsWorkplace.java:2002)
                at org.opencms.workplace.CmsWorkplace.<init>(CmsWorkplace.java:251)
                at org.opencms.workplace.tools.CmsToolDialog.<init>(CmsToolDialog.java:96)
                at org.opencms.workplace.CmsDialog.<init>(CmsDialog.java:283)
                at org.opencms.workplace.CmsWidgetDialog.<init>(CmsWidgetDialog.java:146)
                at org.opencms.workplace.tools.workplace.rfsfile.A_CmsRfsFileWidgetDialog.<init>(A_CmsRfsFileWidgetDialog.java:69)
                at org.opencms.workplace.tools.workplace.rfsfile.CmsRfsFileViewDialog.<init>(CmsRfsFileViewDialog.java:65)
                at org.opencms.workplace.tools.workplace.rfsfile.CmsRfsFileViewDialog.<init>(CmsRfsFileViewDialog.java:78)
                at org.apache.jsp.WEB_002dINF.jsp.offline.system.workplace.admin.workplace.logfileview.index_jsp._jspService(index_jsp.java:62)
                at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
                at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:419)
                at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:389)
                at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:333)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:304)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
                at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:684)
                at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:471)
                at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:402)
                at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:329)
                at org.opencms.flex.CmsFlexRequestDispatcher.forward(CmsFlexRequestDispatcher.java:111)
                at org.opencms.loader.CmsJspLoader.load(CmsJspLoader.java:522)
                at org.opencms.loader.CmsResourceManager.loadResource(CmsResourceManager.java:1199)
                at org.opencms.main.OpenCmsCore.showResource(OpenCmsCore.java:1719)
                at org.opencms.main.OpenCmsServlet.doGet(OpenCmsServlet.java:153)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:621)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:304)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
                at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:684)
                at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:471)
                at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:402)
                at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:329)
                at org.opencms.flex.CmsFlexRequestDispatcher.forward(CmsFlexRequestDispatcher.java:111)
                at org.opencms.util.CmsRequestUtil.forwardRequest(CmsRequestUtil.java:469)
                at org.opencms.workplace.tools.CmsToolManager.jspForwardPage(CmsToolManager.java:475)
                at org.opencms.workplace.administration.CmsAdminDialog.displayDialog(CmsAdminDialog.java:92)
                at org.apache.jsp.WEB_002dINF.jsp.offline.system.workplace.views.admin.admin_002dmain_jsp._jspService(admin_002dmain_jsp.java:63)
                at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
                at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:419)
                at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:389)
                at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:333)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:304)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
                at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:684)
                at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:471)
                at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:402)
                at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:329)
                at org.opencms.loader.CmsJspLoader.load(CmsJspLoader.java:522)
                at org.opencms.loader.CmsResourceManager.loadResource(CmsResourceManager.java:1199)
                at org.opencms.main.OpenCmsCore.showResource(OpenCmsCore.java:1719)
                at org.opencms.main.OpenCmsServlet.doGet(OpenCmsServlet.java:153)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:621)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:304)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
                at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:240)
                at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:164)
                at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:462)
                at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:164)
                at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:100)
                at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118)
                at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:403)
                at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:301)
                at org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:162)
                at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:309)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
```

And the log file is still empty...
I tested OpenCms.getWorkpalceManager().getRFSFileViewer(), and its getFilePath() and getRootPath() are null, and it's frozen.

I have older versions on OpenCms on my Tomcat (I start Tomcat with only one install each time), and they work well.

**Comment by veggie** [comment]
Does Tomcat habe write permissions in the log files folder?

Viele Grüße
Kai
Am 04.09.2015 2:33 nachm. schrieb "SandrineProusteau" <
notifications@github.com>:

**Comment by SandrineProusteau** [comment]
I think so, I didn't change my install methods... and the "setup.log" is filled (3739Ko).

**Comment by gallardo** [comment]
Double check file permissions: OpenCms is complaining about `opencms.log` file (by default, under  `${opencms.WEB-INF}/logs/opencms.log`). Notice that it is a different file. By the way, do you have enough storage space?

**Comment by SandrineProusteau** [comment]
I know it's the "opencms.log" I need. This is the one which is empty (0Ko). There is no special permissions on it, I'm admin of my computer.
I have 78 Go free on my disc.

**Comment by gallardo** [comment]
Uhmmm... In the lists there were some mention to someone missing
```
log4j.appender.OC.File=${opencms.logfile}
```

in `WEB-INF/classes/log4j.properties`. I didn't pay much attention to those threads, though, but you could also check that this _default_ setting has indeed not changed...

**Comment by SandrineProusteau** [comment]
I've already checked it. Nothing changed.

**Comment by gallardo** [comment]
> I tested OpenCms.getWorkpalceManager().getRFSFileViewer(), and its getFilePath() and getRootPath() are null, and it's frozen

And for reference, the line throwing the NPE:

    m_rootPath = new File(OpenCms.getSystemInfo().getLogFileRfsPath()).getParent();

being `new File(...)` what breaks. So the `logFileRfsPath` is `null`...

Some ideas after taking a look at [`org.opencms.main.CmsLog`](https://github.com/alkacon/opencms-core/blob/build_9_5_0/src/org/opencms/main/CmsLog.java):

1. Do you have an `opencms.tld` in your webapp root folder (the parent of `WEB-INF`)? This could potentially result in your described problem. (I would even suggest filing a bug report if this is your case, as this is extremely confusing!)
2. Do you have a `log4j.properties` somewhere conflicting with `WEB-INF/classes`, that the classloader could load first? (I think this is very unlikely, but just in case...)
3. There is an `opencms.set.logfile` property in `log4j.properties` that should also be set to `true`. Is this your case?

**Comment by SandrineProusteau** [comment]
I found that there are files "opencms_init.log" and "opencms-search.log" at the root of my Tomcat7 folder (not in bin/, at the root), and there are updated when I login in the workplace. I don't understand why they are here and not in my "WEB-INF/logs/" folder.

**Comment by gallardo** [comment]
What about your `CATALINA_HOME` and `CATALINA_BASE` dirs? Are they set appropriately?

**Comment by SandrineProusteau** [comment]
I found the origin of the issue : I have spaces in my project folders, and they are not well read in org.opencms.main.CmsLog since modifications of commit 4a96ebdab132b75cd9523d62adaf13df2b082da7 "Improved logging initialization to avoid file not found exceptions". So the "opencms.logfile" and "opencms.logfolder" are not overwritten by OpenCms when started, and stay in my Tomcat folder.
I've modified my resources names in order to make it work now, but it would be great if this regression could be corrected.
